tests/Unit/ParallelAlgorithms/ApparentHorizonFinder/Callbacks/Test_ObserveFieldsAndTimeSeriesOnHorizon.cpp

```diff
@@ -0,0 +1,485 @@
+    using temporal_id_tag = ::Tags::TimeAndPrevious<0>;
+    using frame = Fr;
+
+    using horizon_find_callbacks =
+        tmpl::list<ah::callbacks::ObserveTimeSeriesOnHorizon<
+            tmpl::list<gr::surfaces::Tags::AreaCompute<frame>>, HorizonAB>>;
+    using horizon_find_failure_callbacks = tmpl::list<>;
+
+    using compute_tags_on_element = tmpl::list<>;
+
+    static constexpr ah::Destination destination =
+        ah::Destination::ControlSystem;
+
+    static std::string name() {
+      return "Horizon"s + (std::is_same_v<Fr, Frame::Inertial> ? "A"s : "B"s);
+    }
+  };
+
+  using HorizonA = HorizonAB<Frame::Inertial>;
+  using HorizonB = HorizonAB<Frame::Grid>;
+
+  struct HorizonC : tt::ConformsTo<ah::protocols::HorizonMetavars> {
+    using temporal_id_tag = ::Tags::TimeAndPrevious<0>;
+    using frame = Frame::Inertial;
+
+    using horizon_find_callbacks =
+        tmpl::list<ah::callbacks::ObserveTimeSeriesOnHorizon<
+            tmpl::list<ylm::Tags::MaxRicciScalarCompute>, HorizonC>>;
+    using horizon_find_failure_callbacks = tmpl::list<>;
+
+    using compute_tags_on_element = tmpl::list<>;
+
+    static constexpr ah::Destination destination =
+        ah::Destination::ControlSystem;
+
+    static std::string name() { return "HorizonC"; }
+  };
+
+  struct HorizonD : tt::ConformsTo<ah::protocols::HorizonMetavars> {
+    using temporal_id_tag = ::Tags::TimeAndPrevious<0>;
+    using frame = Frame::Inertial;
+
+    using horizon_find_callbacks =
+        tmpl::list<ah::callbacks::ObserveFieldsOnHorizon<
+            tmpl::list<ylm::Tags::RicciScalar>, HorizonD>>;
+    using horizon_find_failure_callbacks = tmpl::list<>;
+
+    using compute_tags_on_element = tmpl::list<>;
+
+    static constexpr ah::Destination destination = ah::Destination::Observation;
+
+    static std::string name() { return "HorizonD"; }
+  };
+
+  using observed_reduction_data_tags = tmpl::list<>;
+
+  using component_list = tmpl::list<MockObserverWriter<MockMetavariables>,
+                                    MockComponent<MockMetavariables, HorizonA>,
+                                    MockComponent<MockMetavariables, HorizonB>,
+                                    MockComponent<MockMetavariables, HorizonC>,
+                                    MockComponent<MockMetavariables, HorizonD>>;
+};
+
+void run_test() {
+  // Check if either file generated by this test exists and remove them
+  // if so. Check for both files existing before the test runs, since
+  // both files get written when evaluating the list of post interpolation
+  // callbacks below.
+  const std::string h5_file_prefix = "Test_ObserveFieldsAndTimeSeriesOnHorizon";
+  const auto h5_file_name = h5_file_prefix + ".h5";
+  if (file_system::check_if_file_exists(h5_file_name)) {
+    file_system::rm(h5_file_name, true);
+  }
+  const auto surfaces_file_prefix = "Surfaces";
+  if (file_system::check_if_file_exists(surfaces_file_prefix + ".h5"s)) {
```

Comment:
[optional] Use the same style for the surfaces file variables as for the main h5 file, instead of concatenating a `const char*` and `std::string` literal everywhere you need the file name.